drivers/sensor/lps22df/lps22df.c

```diff
@@ -0,0 +1,356 @@
+				enum sensor_channel chan)
+{
+	struct lps22df_data *data = dev->data;
+	const struct lps22df_config * const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps22df_data_t raw_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (lps22df_data_get(ctx, &raw_data) < 0) {
+		LOG_DBG("Failed to read sample");
+		return -EIO;
+	}
+
+	data->sample_press = raw_data.pressure.raw;
+	data->sample_temp = raw_data.heat.raw;
+
+	return 0;
+}
+
+static inline void lps22df_press_convert(struct sensor_value *val,
+					 int32_t raw_val)
+{
+	int32_t press_tmp = raw_val >> 8; /* raw value is left aligned (24 msb) */
+
+	/* Pressure sensitivity is 4096 LSB/hPa */
+	/* Also convert hPa into kPa */
+
+	val->val1 = press_tmp / 40960;
+
+	/* For the decimal part use (3125 / 128) as a factor instead of
+	 * (1000000 / 40960) to avoid int32 overflow
+	 */
+	val->val2 = (press_tmp % 40960) * 3125 / 128;
+}
+
+static inline void lps22df_temp_convert(struct sensor_value *val,
+					int16_t raw_val)
+{
+	/* Temperature sensitivity is 100 LSB/deg C */
+	val->val1 = raw_val / 100;
+	val->val2 = ((int32_t)raw_val % 100) * 10000;
+}
+
+static int lps22df_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct lps22df_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PRESS) {
+		lps22df_press_convert(val, data->sample_press);
+	} else if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		lps22df_temp_convert(val, data->sample_temp);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const uint16_t lps22df_map[] = {0, 1, 4, 10, 25, 50, 75, 100, 200};
+
+static int lps22df_odr_set(const struct device *dev, uint16_t freq)
+{
+	int odr;
+
+	for (odr = 0; odr < ARRAY_SIZE(lps22df_map); odr++) {
+		if (freq == lps22df_map[odr]) {
+			break;
+		}
+	}
+
+	if (odr == ARRAY_SIZE(lps22df_map)) {
+		LOG_DBG("bad frequency");
+		return -EINVAL;
+	}
+
+	if (lps22df_set_odr_raw(dev, odr) < 0) {
+		LOG_DBG("failed to set sampling rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int lps22df_attr_set(const struct device *dev,
+			    enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_set() not supported on this channel.");
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return lps22df_odr_set(dev, val->val1);
+	default:
+		LOG_DBG("operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api lps22df_driver_api = {
+	.attr_set = lps22df_attr_set,
+	.sample_fetch = lps22df_sample_fetch,
+	.channel_get = lps22df_channel_get,
+#if CONFIG_LPS22DF_TRIGGER
+	.trigger_set = lps22df_trigger_set,
+#endif
+};
+
+static int lps22df_init_chip(const struct device *dev)
+{
+	const struct lps22df_config * const cfg = dev->config;
+	struct lps22df_data *data = dev->data;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
```

Comment:
same

Reply:
Yeah, I'll think about changing the STdC APIs for all modules sometimes in the future. After that I should be able to get rid of this casting.